Configure fails on Mac OS X

Hi,

With changes introduced in 60a5636b46ad15ef8c78def67d1ae6d9ba6f7fe5 configure fails with following error:
```
checking for c++11 atomic support without GNU Atomic library... no
checking for main in -latomic... no
configure: error: Please install the GNU Atomic library
```

Relevant part of config.log:
```
configure:19916: checking for c++11 atomic support without GNU Atomic library
configure:19939: g++ -o conftest -std=gnu++0x -g -O2   conftest.cpp -levent -ldouble-conversion -lssl -lgflags -lglog  >&5
conftest.cpp:62:11: error: redefinition of 'main'
      int main() {
          ^
conftest.cpp:55:1: note: previous definition is here
main ()
^
1 error generated.
configure:19939: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME "folly"
| #define PACKAGE_TARNAME "folly"
| #define PACKAGE_VERSION "57.0"
| #define PACKAGE_STRING "folly 57.0"
| #define PACKAGE_BUGREPORT "folly@fb.com"
| #define PACKAGE_URL ""
| #define PACKAGE "folly"
| #define VERSION "57.0"
| #define STDC_HEADERS 1
| #define HAVE_SYS_TYPES_H 1
| #define HAVE_SYS_STAT_H 1
| #define HAVE_STDLIB_H 1
| #define HAVE_STRING_H 1
| #define HAVE_MEMORY_H 1
| #define HAVE_STRINGS_H 1
| #define HAVE_INTTYPES_H 1
| #define HAVE_STDINT_H 1
| #define HAVE_UNISTD_H 1
| #define HAVE_DLFCN_H 1
| #define LT_OBJDIR ".libs/"
| #define HAVE_STDCXX_0X /**/
| #define HAVE_LIBGLOG 1
| #define HAVE_LIBGFLAGS 1
| #define HAVE_LIBSSL 1
| #define HAVE_BOOST /**/
| #define HAVE_BOOST_CONTEXT /**/
| #define HAVE_BOOST_PROGRAM_OPTIONS /**/
| #define HAVE_BOOST_THREAD /**/
| #define HAVE_BOOST_REGEX /**/
| #define HAVE_BOOST_SYSTEM /**/
| #define HAVE_BOOST_FILESYSTEM /**/
| #define STDC_HEADERS 1
| #define HAVE_FCNTL_H 1
| #define HAVE_INTTYPES_H 1
| #define HAVE_LIMITS_H 1
| #define HAVE_STDINT_H 1
| #define HAVE_STDLIB_H 1
| #define HAVE_STRING_H 1
| #define HAVE_SYS_TIME_H 1
| #define HAVE_UNISTD_H 1
| #define HAVE_LIBDOUBLE_CONVERSION 1
| #define HAVE_LIBEVENT 1
| #define HAVE_STDBOOL_H 1
| #define TIME_WITH_SYS_TIME 1
| #define HAVE_INT128_T 1
| #define HAVE_PTRDIFF_T 1
| #define FINAL final
| #define OVERRIDE override
| #define HAVE_STD__THIS_THREAD__SLEEP_FOR 1
| #define USE_LIBCPP 1
| /* end confdefs.h.  */
| 
| int
| main ()
| {
| 
|   ;
|   return 0;
| }
|       #include <atomic>
|       int main() {
|         struct Test { int val; };
|         std::atomic<Test> s;
|         s.is_lock_free();
|       }
| 
configure:19947: result: no
configure:19951: checking for main in -latomic
configure:19970: g++ -o conftest -std=gnu++0x -g -O2   conftest.cpp -latomic  -levent -ldouble-conversion -lssl -lgflags -lglog  >&5
ld: library not found for -latomic
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

--- a/folly/configure.ac
+++ b/folly/configure.ac
@@ -235,7 +235,7 @@ AC_CACHE_CHECK(
   [for c++11 atomic support without GNU Atomic library],
   [folly_cv_lib_libatomic],
   [AC_LINK_IFELSE(
-    [AC_LANG_PROGRAM[
+    [AC_LANG_SOURCE[
       #include <atomic>
       int main() {
         struct Test { int val; };

